typescript/sdk/src/token/adapters/EvmTokenAdapter.ts

```diff
@@ -762,3 +766,268 @@ export class EvmXERC20VSAdapter
     });
   }
 }
+
+export class EvmKhalaniIntentTokenAdapter
```

Comment:
All the communication between HL and Khalani happens in this adapter.

Besides the `approve` and `deposit` TXs, Khalani creates a `refine` from the user inputs and returns an object with the bridge stats. This object is then signed by the user and used to create an intent, which is forwarded to the network of solvers.